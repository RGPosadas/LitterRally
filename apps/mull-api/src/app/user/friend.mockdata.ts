import { RegistrationMethod } from '@mull/types';
import { mockAllDirectMessageChannels } from '../channel/channel.mockdata';
import { Friend } from '../entities';
import { mockAllPosts } from '../post/post.mockdata';

// Friends of mockAllUsers[0]
export const mockAllFriends: Friend[] = [
  {
    id: 7,
    password: 'password',
    email: 'me@me.com',
    timezone: '',
    name: 'Cristian',
    dob: new Date(),
    description: "There's a first for everything",
    friends: [],
    tokenVersion: 0,
    registrationMethod: RegistrationMethod.LOCAL,
    avatar: null,
    latestPost: mockAllPosts[3],
    directMessageChannel: mockAllDirectMessageChannels[0],
  },
  {
    id: 12,
    password: 'abc123',
    email: 'tim@green.co',
    timezone: '',
    name: 'Tim',
    dob: new Date(),
    description: '',
    friends: [],
    tokenVersion: 0,
    registrationMethod: RegistrationMethod.LOCAL,
    avatar: null,
    latestPost: null,
    directMessageChannel: null,
  },
];
