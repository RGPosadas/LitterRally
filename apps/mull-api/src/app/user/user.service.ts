import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository, Connection } from 'typeorm';
import { User } from '../entity';

@Injectable()
export class UserService {
  constructor(
    @InjectRepository(User)
    private userRepository: Repository<User>,
    private connection: Connection
  ) {}

  findAll(): Promise<User[]> {
    return this.userRepository.find();
  }

  findOne(id: number): Promise<User> {
    return this.userRepository.findOne(id);
  }

  async findAllFriends(id: number): Promise<User[]> {
    const what = await this.userRepository.find({ relations: ['friends'] });
    console.log(JSON.stringify(what));
    return null;
  }

  async remove(id: string): Promise<void> {
    await this.userRepository.delete(id);
  }

  async createOne(user: User) {
    const queryRunner = this.connection.createQueryRunner();
    await queryRunner.connect();
    await queryRunner.startTransaction();
    try {
      await queryRunner.manager.save(User, user);
      await queryRunner.commitTransaction();
    } catch (err) {
      await queryRunner.rollbackTransaction();
    } finally {
      await queryRunner.release();
    }
  }
}
