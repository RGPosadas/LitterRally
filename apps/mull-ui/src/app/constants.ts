export const ROUTES = {
  HOME: '/home',
  MAP: '/map',
  NEW_EVENT: '/new-event',
  TOOLS: '/tools',
  PROFILE: '/profile',
};
