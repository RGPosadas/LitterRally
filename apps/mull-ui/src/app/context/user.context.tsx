import React from 'react';

const UserContext = React.createContext<{
  userId: number;
  setUserId: React.Dispatch<React.SetStateAction<number>>;
  accessToken: string;
  setAccessToken: React.Dispatch<React.SetStateAction<string>>;
}>(null);

export const UserProvider = UserContext.Provider;
export const UserConsumer = UserContext.Consumer;

export default UserContext;
