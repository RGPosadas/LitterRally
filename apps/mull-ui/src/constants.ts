import { EventRestriction, IEvent } from '@mull/types';

/**
 * enum of the current routes for the application
 */
export enum ROUTES {
  LOGIN = '/login',
  HOME = '/home',
  MAP = '/map',
  CREATE_EVENT = '/create-event',
  TOOLS = '/tools',
  MESSAGES = '/messages',
  PROFILE = '/profile',
}

export const DAY_IN_MILLISECONDS = 86400000;

export const dummyEvent: IEvent = {
  id: 1,
  title: 'Test title',
  description:
    'Lorem ipsum dolor sit amet, regione invidunt democritum vim in, movet antiopam gubergren ne per. Est nibh magna scribentur ea, vel te munere deseruisse disputationi, eros meis ludus ne sea. Odio prompta legendos in sea, ut mei scripta labores theophrastus, id molestie probatus periculis mea. Noluisse invenire splendide sed ne, at erat quando laudem nec, possim apeirian vix at. At eam animal efficiendi interpretaris, eirmod offendit adversarium per et, summo qualisque efficiendi in has. Vel detraxit accusata ea.',
  startDate: new Date(2000, 0, 1, 0, 0, 0, 0),
  endDate: new Date(2000, 0, 2, 0, 0, 0, 0),
  location: {
    id: 1,
    point: '1260 Remembrance Road Montreal, Qc',
  },
  restriction: EventRestriction.NONE,
  participants: [
    {
      name: 'User 1',
    },
    {
      name: 'User 2',
    },
  ],
  host: {
    name: 'Test user',
  },
};
