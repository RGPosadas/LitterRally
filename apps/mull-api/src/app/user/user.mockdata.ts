import { RegistrationMethod } from '@mull/types';
import { User } from '../entities';
import { CreateUserInput, UpdateUserInput } from './inputs/user.input';

export const mockPartialUser: CreateUserInput | UpdateUserInput = {
  password: 'password',
  email: 'mock@mock.com',
  dob: new Date(),
  name: 'Mock McMockson',
  registrationMethod: RegistrationMethod.LOCAL,
};

export const mockNewPartialUser: CreateUserInput = {
  password: 'password',
  email: 'mock2@mock.com',
  dob: new Date(),
  name: 'Mock McMockson Jr.',
  registrationMethod: RegistrationMethod.LOCAL,
};

export const mockAllUsers: User[] = [
  {
    id: 1,
    password: '$2b$10$Wo/ClOmaI/qItblWM1saEeWgqv9S9nt6QO4KW1kBXdzPT1.q8XuuG',
    email: 'mock@mock.com',
    timezone: '',
    name: 'Bob',
    dob: new Date(),
    description: 'I am very good programmer',
    friends: [
      {
        id: 7,
        password: 'password',
        email: 'me@me.com',
        timezone: '',
        name: 'Cristian',
        dob: new Date(),
        description: "There's a first for everything",
        friends: [],
        tokenVersion: 0,
        registrationMethod: RegistrationMethod.LOCAL,
      },
      {
        id: 12,
        password: 'abc123',
        email: 'tim@green.co',
        timezone: '',
        name: 'Tim',
        dob: new Date(),
        description: '',
        friends: [],
        tokenVersion: 0,
        registrationMethod: RegistrationMethod.LOCAL,
      },
    ],
    tokenVersion: 0,
    registrationMethod: RegistrationMethod.LOCAL,
  },
  {
    id: 7,
    password: 'password',
    email: 'me@me.com',
    timezone: '',
    name: 'Cristian',
    dob: new Date(),
    description: "There's a first for everything",
    friends: [
      {
        id: 1,
        password: 'abc123',
        email: 'gr@ph.ql',
        timezone: '',
        name: 'Bob',
        dob: new Date(),
        description: 'I am very good programmer',
        friends: [],
        tokenVersion: 0,
        registrationMethod: RegistrationMethod.LOCAL,
      },
      {
        id: 12,
        password: 'abc123',
        email: 'tim@green.co',
        timezone: '',
        name: 'Tim',
        dob: new Date(),
        description: '',
        friends: [],
        tokenVersion: 0,
        registrationMethod: RegistrationMethod.LOCAL,
      },
    ],
    tokenVersion: 0,
    registrationMethod: RegistrationMethod.GOOGLE,
  },
  {
    id: 3,
    password: 'password',
    email: 'me@me.com',
    timezone: '',
    name: 'Jose',
    dob: new Date(),
    description: 'What goes up, comes down',
    friends: [
      {
        id: 1,
        password: 'abc123',
        email: 'gr@ph.ql',
        timezone: '',
        name: 'Bob',
        dob: new Date(),
        description: 'I am very good programmer',
        friends: [],
        registrationMethod: RegistrationMethod.LOCAL,
        tokenVersion: 0,
      },
      {
        id: 12,
        password: 'abc123',
        email: 'tim@green.co',
        timezone: '',
        name: 'Tim',
        dob: new Date(),
        description: '',
        friends: [],
        registrationMethod: RegistrationMethod.LOCAL,
        tokenVersion: 0,
      },
    ],
    registrationMethod: RegistrationMethod.GOOGLE,
    tokenVersion: 0,
  },
];
