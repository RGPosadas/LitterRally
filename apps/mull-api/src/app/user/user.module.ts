import { Module } from '@nestjs/common';
import { EntitiesModule } from '../entities';
import { EventService } from '../event/event.service';
import { MediaService } from '../media/media.service';
import { UserResolver } from './user.resolver';
import { UserService } from './user.service';

@Module({
  imports: [EntitiesModule],
  providers: [UserResolver, UserService, EventService, MediaService],
  exports: [UserService],
})
export class UserModule {}
