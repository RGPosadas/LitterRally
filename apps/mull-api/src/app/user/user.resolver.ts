import { RegistrationMethod } from '@mull/types';
import { UnauthorizedException } from '@nestjs/common';
import { Args, Int, Mutation, Parent, Query, ResolveField, Resolver } from '@nestjs/graphql';
import { genSalt, hash } from 'bcrypt';
import { User } from '../entities';
import { CreateUserInput, UpdateUserInput } from './inputs/user.input';
import { UserService } from './user.service';

@Resolver(/* istanbul ignore next */ () => User)
export class UserResolver {
  constructor(private readonly userService: UserService) {}

  @Query(/* istanbul ignore next */ () => [User])
  async users() {
    return this.userService.findAll();
  }

  @Query(/* istanbul ignore next */ () => User)
  async user(@Args('id', { type: /* istanbul ignore next */ () => Int }) id: number) {
    return this.userService.findOne(id);
  }

  @ResolveField(/* istanbul ignore next */ () => [User])
  async friends(@Parent() user: User) {
    const { id } = user;
    return this.userService.findAllFriends(id);
  }

  @Mutation(/* istanbul ignore next */ () => User)
  async createUser(@Args('createUserInput') createUserInput: CreateUserInput) {
    if (createUserInput.registrationMethod === RegistrationMethod.LOCAL) {
      const salt = await genSalt(10);
      const hashed = await hash(createUserInput.password, salt);
      createUserInput.password = hashed;
    }
    const existingUser = await this.userService.findUnique(
      createUserInput.email,
      createUserInput.registrationMethod
    );
    if (existingUser.length > 0)
      throw new UnauthorizedException('User with this email already exists.');
    return this.userService.create(createUserInput);
  }

  @Mutation(/* istanbul ignore next */ () => User)
  async updateUser(@Args('updateUserInput') updateUserInput: UpdateUserInput) {
    return this.userService.updateUser(updateUserInput);
  }

  @Mutation(/* istanbul ignore next */ () => User)
  async deleteUser(@Args('id', { type: /* istanbul ignore next */ () => Int }) id: number) {
    return this.userService.delete(id);
  }
}
