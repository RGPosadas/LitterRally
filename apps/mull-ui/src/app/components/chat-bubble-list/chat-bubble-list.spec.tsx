import { render } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import { mockPosts } from '../../../constants';
import { UserProvider } from '../../context/user.context';
import ChatBubbleList from './chat-bubble-list';

describe('ChatBubble', () => {
  it('should render successfully', () => {
    const { baseElement } = render(
      <UserProvider value={{ userId: 1, setUserId: jest.fn() }}>
        <ChatBubbleList posts={mockPosts} />
      </UserProvider>
    );

    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const tree = renderer
      .create(
        <UserProvider value={{ userId: 1, setUserId: jest.fn() }}>
          <ChatBubbleList posts={mockPosts} />
        </UserProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
