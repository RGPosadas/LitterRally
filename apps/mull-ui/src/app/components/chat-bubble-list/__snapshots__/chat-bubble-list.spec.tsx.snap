// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatBubble should match snapshot 1`] = `
<div
  className="chat-bubble-list"
>
  <div
    className="chat-container"
  >
    <p
      className="announcement-time"
    >
      22:33 2021-02-28
    </p>
    <div
      className="current-user-chat-container"
    >
      <img
        alt="user"
        className="user-picture"
        src="http://localhost:3333/api/media/1"
      />
      <p
        className="chat-bubble"
      >
        message 1 from user 1, current user id is 1
      </p>
    </div>
  </div>
  <div
    className="chat-container"
  >
    <p
      className="announcement-time"
    >
      22:33 2021-02-28
    </p>
    <div
      className="other-user-chat-container"
    >
      <img
        alt="user"
        className="user-picture"
        src="http://localhost:3333/api/media/1"
      />
      <p
        className="chat-bubble"
      >
        message 2 from user 2
      </p>
    </div>
  </div>
  <div
    className="chat-container"
  >
    <p
      className="announcement-time"
    >
      22:33 2021-02-28
    </p>
    <div
      className="other-user-chat-container"
    >
      <img
        alt="user"
        className="user-picture"
        src="http://localhost:3333/api/media/1"
      />
      <p
        className="chat-bubble"
      >
        message 3 from user 3
      </p>
    </div>
  </div>
</div>
`;
